### PR TITLE
Allow theming for Accordion, Calendar and Carousel stories - Issue 5735 

### PR DESCRIPTION
**Pull request by matheushahnn**
What does this PR do?
Working on #5735, theming support for following components is added:

- [x] Accordion
- [x] Calendar
- [x] Carousel 

The theme can now be switched between the currently available themes: base, grommet, hpe.

Where should the reviewer start?
Start in src/js/components and look at the changes made to the following folders:

* Accordion
* Calendar
* Carousel

What testing has been done on this PR?
Passed the existing tests ran through jest and manually verified the changes on storybook.

How should this be manually tested?
Run storybook and navigate to the following stories: Accordion, Calendar, Carousel. Themes can now be switched for these stories using the Theme button.

What are the relevant issues?
#5737

Do the grommet docs need to be updated?
No.

Should this PR be mentioned in the release notes?
No.

Is this change backwards compatible or is it a breaking change?
Not a breaking change.

### CODE REVIEW

**Comment by matheushahnn** [comment]
In Card/Simple the **hpe** theme doesn't fits weel
image

I believe it is related to padding style added in header, body and footer in **hpe** theme.
https://github.com/grommet/grommet-theme-hpe/blob/master/src/js/themes/hpe.js#L336

File: src/js/components/Accordion/stories/CustomHeader.js [hunk 1]
```diff
@@ -23,10 +15,12 @@ const CustomHeaderAccordion = () => {
   const [activeIndex, setActiveIndex] = useState([0]);
 
   return (
-    <Grommet theme={grommet}>
+    // Uncomment <Grommet> lines when using outside of storybook
+    // <Grommet theme={...}>
+    <Box>
```
Comment (jcfilben) [inline src/js/components/Accordion/stories/CustomHeader.js:20]: We shouldn't need to wrap with a Box here

File: src/js/components/Calendar/stories/CustomDay.js [hunk 2]
```diff
@@ -60,14 +53,15 @@ export const CustomDayCalendar = () => {
         <Calendar onSelect={onSelect} date={selectedDay} fill>
           {({ day, isSelected }) => (
             <Box background={isSelected ? 'light-3' : 'white'} border fill>
-              <Box pad="medium" align="center" justify="center" fill>
+              <Box pad="small" align="center" justify="center" fill>
```
Comment (jcfilben) [inline src/js/components/Calendar/stories/CustomDay.js:56]: Why is the pad changing from medium to small here?

File: src/js/components/Accordion/stories/CustomThemed/Custom.js [hunk 3]
```diff
@@ -29,14 +29,7 @@ const customAccordionTheme = {
...
     border: undefined,
-    panel: {
-      // border: {
-      //   side: 'horizontal',
-      //   size: 'medium',
-      //   color: '#DADADA',
-      //   style: 'dotted',
-      // },
-    },
+    panel: {},
```
Comment (taysea) [inline src/js/components/Accordion/stories/CustomThemed/Custom.js:32]: If this is an empty object, should we just remove it?

File: src/js/components/Card/stories/Simple.js [hunk 4]
```diff
@@ -1,42 +1,34 @@
...
-      {/* children that are not Card internal components */}
-      <Card pad="small" gap="medium" background="light-1">
-        <Text>text - random component</Text>
-        <Box>box - random component</Box>
-      </Card>
-    </Box>
-  </Grommet>
+  // Uncomment <Grommet> lines when using outside of storybook
+  // <Grommet theme={...}>
+  <Box pad="large" gap="medium" height="large" width="medium">
```
Comment (taysea) [inline src/js/components/Card/stories/Simple.js:8]:
```suggestion
  <Box pad="large" gap="medium" width="medium">
```
Think we should remove this fixed height and let the cards take the vertical height that they need.

Comment (taysea) [reply src/js/components/Card/stories/Simple.js:8]: This should resolve the issue we see on the story when using hpe theme.

File: src/js/components/Calendar/stories/CustomDay.js (hunk 2 above)
Comment (matheushahnn) [reply src/js/components/Calendar/stories/CustomDay.js:56]: When I opened the storybook I saw two differents size calendars, by the title I assume the only difference should be the onSelect set or not, so I changed to keep both calendars with the same height.

File: src/js/components/Accordion/stories/CustomThemed/Custom.js (hunk 3 above)
Comment (matheushahnn) [reply src/js/components/Accordion/stories/CustomThemed/Custom.js]: Removed

File: src/js/components/Card/stories/Simple.js (hunk 4 above)
Comment (matheushahnn) [reply src/js/components/Card/stories/Simple.js]: Solved the problem. Thank you

**Review by jcfilben — APPROVED** [review]
Looks good

**Pull request merged**